test/parallel/test-util-promisify.js

```diff
@@ -7,6 +7,18 @@ const vm = require('vm');
 const { promisify } = require('util');
 const { customPromisifyArgs } = require('internal/util');
 
+common.expectWarning('Warning', 'Calling promisify on async functions is likely a mistake.');
+promisify(async () => {});
+
+{
+  const warningHandler = common.mustNotCall();
+  process.on('warning', warningHandler);
+  function foo() {}
+  foo.constructor = 'AsyncFunction';
```

Comment:
```suggestion
  foo.constructor = { name: 'AsyncFunction' };
```
This might be more practical, as it ensures that the check is not loosened to common `original.constructor.name !== 'AsyncFunction'`.